protocol/messenger_pairing_and_syncing_test.go

```diff
@@ -58,12 +59,127 @@ func (s *MessengerPairingSuite) TestEnableNonExistingInstallation() {
 
 }
 
-// TestMessengerPairAfterSeedPhrase tests the scenario where alice2 wants to sync with alice1
+type mockEnableInstallationAndPair struct {
+	installationID                string
+	getInstallationIDInvokedFirst bool
+}
+
+func (m *mockEnableInstallationAndPair) GetInstallationID() string {
+	if !m.getInstallationIDInvokedFirst {
+		m.getInstallationIDInvokedFirst = true
+		return m.installationID
+	}
+	return "wrong installation ID"
+}
+
+func (m *mockEnableInstallationAndPair) Validate() error {
+	return nil
+}
```

Comment:
Why not generate a mock instead? 🤔

Reply:
Nitpick: technically I think this is more like a stub than a mock. Mocks have other abilities built in, like tracking call count, tracking arguments, and in general other behavioral checks, like auto-implementing functions to return nil by default.

Reply:
I would say so @qfrank. With a mock we usually write the expectations even before the test runs, saying for example, how many times a specific call is expected to happen, or dynamically making the mock respond to certain input. If we implement an interface and provide static output like what this PR does, then it's the perfect definition of a stub.

Sometimes it's hard to differentiate and name different test doubles. Sometimes it's a gray area.